src/corefx/System.Globalization.Native/locale.cpp

```diff
@@ -0,0 +1,373 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+#include <stdint.h>
+#include <string.h>
+
+#include "localeInformationConstants.hpp"
+
+#include "unicode/dcfmtsym.h" //decimal
+#include "unicode/dtfmtsym.h" //date
+#include "unicode/ulocdata.h"
+
+/*
+Function:
+GetLocale
+
+Returns a locale given the locale name
+*/
+Locale GetLocale(const UChar* localeName, bool canonize = false)
```

Comment:
Seems like using byte would be a decent optimization since this method is called frequently, however I prefer the encapsulation and thus the conversion. Memory here is stack-based, so should be fast anyway.